### PR TITLE
Drop python3.9 and add python3.12

### PR DESCRIPTION
**Pull request by YooSunYoung**
Dependency updates is done in #42

### CODE REVIEW

**Comment by YooSunYoung** [comment]
@scipp/ess-maintainers
Can someone with owner role in this repository
update the required checks (3.9, py39 -> 3.10, py310)in the settings??
Thanks...!

**Comment by jl-wynen** [comment]
> @scipp/ess-maintainers Can someone with owner role in this repository update the required checks (3.9, py39 -> 3.10, py310)in the settings?? Thanks...!

Done

File: conda/meta.yaml [hunk 1]
```diff
@@ -11,6 +11,7 @@ requirements:
     - setuptools
     - setuptools_scm
   run:
+<<<<<<< before updating
```
Comment (jl-wynen) [inline conda/meta.yaml:14]: Merge conflict

**Review by jl-wynen — CHANGES_REQUESTED** [review]

**Review by jl-wynen — APPROVED** [review]

File: conda/meta.yaml (hunk 1 above)
Comment (jl-wynen) [reply conda/meta.yaml]: This happens to me as well. I think there is some bad interaction between copier's conflict markers and git itself.

**Pull request merged**